src/smc-webapp/editor_chat.cjsx

```diff
@@ -178,6 +178,11 @@ class ChatActions extends Actions
 
         if messages_before != messages
             @setState(messages: messages)
+            store = @redux.getStore(@name)
+            if store.get('saved_position') + store.get('offset') + 20 > store.get('height') or store.get('height') == store.get('client_height')
+                project_id = @name.slice(7,43)
```

Comment:
This seems like a pretty fragile way to get the project_id and path. I suggest passing it to the action or finding it in the store.